src/main/java/ch/jalu/configme/resource/PropertyPathTraverser.java

```diff
@@ -83,41 +81,76 @@ public static class PathElement {
 
         private final int indentationLevel;
         private final String name;
-        private final List<String> comments;
+        private final String fullPath;
         private final boolean isFirstElement;
         private boolean isFirstOfGroup;
+        private boolean isEndOfPath;
 
-        public PathElement(int indentationLevel, @NotNull String name, @NotNull List<String> comments,
+        public PathElement(int indentationLevel, @NotNull String name, @NotNull String fullPath,
                            boolean isFirstElement) {
             this.indentationLevel = indentationLevel;
             this.name = name;
-            this.comments = comments;
+            this.fullPath = fullPath;
             this.isFirstElement = isFirstElement;
         }
 
+        /**
+         * @return the hierarchy level of this path element
+         */
         public int getIndentationLevel() {
             return indentationLevel;
         }
 
+        /**
+         * @return the name of this path element (e.g. "driver")
+         */
         public @NotNull String getName() {
             return name;
         }
 
-        public @NotNull List<String> getComments() {
-            return comments;
+        /**
+         * @return the full path of this element (e.g. "config.datasource.driver")
+         */
+        public @NotNull String getFullPath() {
+            return fullPath;
         }
 
+        /**
+         * @return true if this path element is the <b>very first</b> element returned by the traverser; false otherwise
+         */
         public boolean isFirstElement() {
             return isFirstElement;
         }
 
+        /**
+         * Returns if this path element is the first new path element of a property. For example, if a property
+         * {@code config.datasource.mysql.type} was previously processed and we're now processing the path
+         * {@code config.datasource.driver.version}, then the path element representing {@code driver} is considered
+         * to be the first of the group.
+         *
+         * @return true if this path element is the first new element of the path, false otherwise
+         */
         public boolean isFirstOfGroup() {
             return isFirstOfGroup;
         }
 
         void setFirstOfGroup(boolean firstOfGroup) {
             isFirstOfGroup = firstOfGroup;
         }
-    }
 
+        /**
+         * Returns if this path element is at the end, i.e. whether it represents a leaf path that is associated to
```

Comment:
This sounds weird but I don't know how to better phrase it.
* Returns if this is a final path element, i.e. whether it is the last path element of a property's path?
* Failing that, "associated to" should probably be "associated with"